txscript/script.go

```diff
@@ -766,7 +766,8 @@ func GetWitnessSigOpCount(sigScript, pkScript []byte, witness wire.TxWitness) in
 	}
 	if IsPayToScriptHash(pkScript) &&
 		isPushOnly(sigPops) &&
-		IsPayToWitnessScriptHash(sigScript[1:]) {
+		(IsPayToWitnessScriptHash(sigScript[1:]) ||
```

Comment:
This does not match the same behavior as Core and thus might be able to result in a hard fork.  Its behavior is to count the witness signature operations if the following conditions hold:

* The pkScript is a pay-to-script-hash (check)
* The sigScript only pushes data (check)
* The parsed script (up to the point of failure if there is one) is considered a witness program (**deviation**)

This code, on the other hand, handles the first two conditions properly, but will only count the sigops if the witness program is specifically a pay-to-witness-script-hash or a pay-to-witness-pubkey-hash.  That behavior is not the same.